File: bms/go/go-gt/newHeap/newHeap.go
package test

import (
	"math"
)

var ()

const ()

func newHeap(n int64) (h Heap) {
	h.N = n
	h.I = make([]int64, n)
	h.A = make([]int64, n)
	h.W = make([]int64, n)
	var i int64
	for i = 0; i < n; i++ {
		h.I[i] = i
		h.A[i] = i
		h.W[i] = math.MaxInt64
	}
	return h
}

type Heap struct {
	N int64
	I []int64
	A []int64
	W []int64
}

File: translations_only/fail/go/go-gt/newHeap/mistral/Hinted/newHeap.rs

use std::boxed::Box;
use std::mem;
use std::vec::Vec;

type Int64 = i64;
type Heap = Box<HeapStruct>;

struct HeapStruct {
    n: Int64,
    i: Vec<Int64>,
    a: Vec<Int64>,
    w: Vec<Int64>,
}

const MAX_INT64: Int64 = i64::MAX;

fn new_heap(n: Int64) -> Heap {
    let mut h = Box::new(HeapStruct {
        n: n,
        i: vec![0; n as usize],
        a: vec![0; n as usize],
        w: vec![MAX_INT64; n as usize],
    });
    unsafe {
        for i in 0..n {
            (*h).i[i as usize] = i;
            (*h).a[i as usize] = i;
            (*h).w[i as usize] = MAX_INT64;
        }
    }
    h
}
